### PR TITLE
fix warnings/notes in PEcAn.uncertainty package

### PR DESCRIPTION
**Pull request by moki1202**

### CODE REVIEW

File: modules/uncertainty/R/plots.R [hunk 1]
```diff
@@ -117,11 +117,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
   post.x <- seq(from = min(sa.sample), to = max(sa.sample), length.out = LENGTH_OUT)
   
   saplot <- saplot + ## plot spline function
-    ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = post.x, y = sa.spline(post.x)), size = linesize) + 
+    ggplot2::geom_line(ggplot2::aes(x=post.x, y=sa.spline(post.x)), data = data.frame(x = post.x, y = sa.spline(post.x)), size = linesize) +
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:120]: This isn't doing quite what it looks like -- this makes ggplot go find `post.x` in the enclosing environment and ignore its `data` argument, which has the potential to cause surprises later. I believe you should be able to use `x = .data$x, y = .data$y` here same as in dplyr functions.

File: modules/uncertainty/R/run.sensitivity.analysis.R [hunk 2]
```diff
@@ -121,7 +126,9 @@ run.sensitivity.analysis <-
           settings, "sensitivity.output", "Rdata", all.var.yr = FALSE,
           ensemble.id = ensemble.id, variable = variable.fn,
           start.year = start.year, end.year = end.year)
-        load(fname, envir = environment())
+        env = new.env()
+        load(fname, envir = env)
+        sensitivity.output <- env$sensitivty.output
```
Comment (infotroph) [inline modules/uncertainty/R/run.sensitivity.analysis.R:131]: typo?
```suggestion
        sensitivity.output <- env$sensitivity.output
```

File: modules/uncertainty/tests/Rcheck_reference.log [hunk 3]
```diff
@@ -102,96 +33,23 @@ Maintainer field differs from that derived from Authors@R
...
-runModule.run.ensemble.analysis: no visible global function definition
-  for ‘papply’
-runModule.run.ensemble.analysis: no visible global function definition
-  for ‘is.Settings’
-runModule.run.sensitivity.analysis: no visible global function
-  definition for ‘is.MultiSettings’
-runModule.run.sensitivity.analysis: no visible global function
-  definition for ‘papply’
-runModule.run.sensitivity.analysis: no visible global function
-  definition for ‘is.Settings’
```
Comment (infotroph) [inline modules/uncertainty/tests/Rcheck_reference.log:174]: Feels good to see these go!

File: modules/uncertainty/R/plots.R (hunk 1 above)
Comment (moki1202) [reply modules/uncertainty/R/plots.R:120]: understood! I'll make the necessary changes.

File: modules/uncertainty/R/run.sensitivity.analysis.R (hunk 2 above)
Comment (moki1202) [reply modules/uncertainty/R/run.sensitivity.analysis.R]: resolved.

File: modules/uncertainty/R/plots.R (hunk 1 above)
Comment (moki1202) [reply modules/uncertainty/R/plots.R]: resolved

File: modules/uncertainty/R/plots.R [hunk 4]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
   
   
   trait.plot <- base.plot + ggplot2::ggtitle("Parameter") + 
-    ggplot2::geom_text(ggplot2::aes(y = 1, x = points, label = trait.labels, hjust = 1), size = fontsize$axis/3) + 
+    ggplot2::geom_text(ggplot2::aes(y = 1, x = plot.data$points, label = trait.labels, hjust = 1), size = fontsize$axis/3) +
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:53]:
```suggestion
    ggplot2::geom_text(ggplot2::aes(y = 1, x = .data$points, label = trait.labels, hjust = 1), size = fontsize$axis/3) + 
```

File: modules/uncertainty/R/plots.R [hunk 5]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
   
   
   trait.plot <- base.plot + ggplot2::ggtitle("Parameter") + 
-    ggplot2::geom_text(ggplot2::aes(y = 1, x = points, label = trait.labels, hjust = 1), size = fontsize$axis/3) + 
+    ggplot2::geom_text(ggplot2::aes(y = 1, x = plot.data$points, label = trait.labels, hjust = 1), size = fontsize$axis/3) + 
     ggplot2::scale_y_continuous(breaks = c(0, 0), limits = c(0, 1)) + 
     ggplot2::theme(axis.text.x = ggplot2::element_blank())
   cv.plot <- base.plot + ggplot2::ggtitle("CV (%)") + 
-    ggplot2::geom_pointrange(ggplot2::aes(x = points, y = coef.vars, ymin = 0, ymax = coef.vars), size = 1.25) +
+    ggplot2::geom_pointrange(ggplot2::aes(x = plot.data$points, y = plot.data$coef.vars, ymin = 0, ymax = plot.data$coef.vars), size = 1.25) +
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:57]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$coef.vars, ymin = 0, ymax = .data$coef.vars), size = 1.25) +
```

File: modules/uncertainty/R/plots.R [hunk 6]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
...
     ggplot2::theme(plot.title = ggplot2::element_text(size = fontsize$title))
   
   el.plot <- base.plot + ggplot2::ggtitle("Elasticity") + 
     ggplot2::theme(plot.title = ggplot2::element_text(size = fontsize$title)) + 
-    ggplot2::geom_pointrange(ggplot2::aes(x = points, y = elasticities, ymin = 0, ymax = elasticities), size = 1.25)
+    ggplot2::geom_pointrange(ggplot2::aes(x = plot.data$points, y = plot.data$elasticities, ymin = 0, ymax = plot.data$elasticities), size = 1.25)
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:62]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$elasticities, ymin = 0, ymax = .data$elasticities), size = 1.25)
```

File: modules/uncertainty/R/plots.R [hunk 7]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
...
   
   pv.plot <- base.plot + ggplot2::ggtitle("Variance") + 
     ggplot2::theme(plot.title = ggplot2::element_text(size = fontsize$title)) + 
-    ggplot2::geom_pointrange(ggplot2::aes(x = points, sqrt(variances), ymin = 0, ymax = sqrt(variances)), size = 1.25)
+    ggplot2::geom_pointrange(ggplot2::aes(x = plot.data$points, sqrt(plot.data$variances), ymin = 0, ymax = sqrt(plot.data$variances)), size = 1.25)
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:66]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, sqrt(.data$variances), ymin = 0, ymax = sqrt(.data$variances)), size = 1.25)
```

File: modules/uncertainty/R/plots.R [hunk 8]
```diff
@@ -138,11 +138,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
   if (!is.null(prior.sa.sample) & !is.null(prior.sa.spline)) {
     prior.x <- seq(from = min(prior.sa.sample), to = max(prior.sa.sample), length.out = LENGTH_OUT)
     saplot <- saplot + ## plot spline
-      ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
+      ggplot2::geom_line(ggplot2::aes(x=prior.x, y=prior.sa.spline(prior.x)), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)),
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:141]:
```suggestion
      ggplot2::geom_line(ggplot2::aes(x = .data$x, y = .data$y, data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
```

File: modules/uncertainty/R/plots.R [hunk 9]
```diff
@@ -138,11 +138,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
   if (!is.null(prior.sa.sample) & !is.null(prior.sa.spline)) {
     prior.x <- seq(from = min(prior.sa.sample), to = max(prior.sa.sample), length.out = LENGTH_OUT)
     saplot <- saplot + ## plot spline
-      ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
+      ggplot2::geom_line(ggplot2::aes(x=prior.x, y=prior.sa.spline(prior.x)), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
                 size = linesize, color = "grey") + ## plot points used to evaluate spline
-      ggplot2::geom_point(ggplot2::aes(x, y), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
+      ggplot2::geom_point(ggplot2::aes(x=prior.sa.sample, y=prior.sa.spline(prior.sa.sample)), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)),
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:143]:
```suggestion
      ggplot2::geom_point(ggplot2::aes(x = .data$x, y = .data$y, data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
```

File: modules/uncertainty/R/plots.R [hunk 10]
```diff
@@ -138,11 +138,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
...
     saplot <- saplot + ## plot spline
-      ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
+      ggplot2::geom_line(ggplot2::aes(x=prior.x, y=prior.sa.spline(prior.x)), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
                 size = linesize, color = "grey") + ## plot points used to evaluate spline
-      ggplot2::geom_point(ggplot2::aes(x, y), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
+      ggplot2::geom_point(ggplot2::aes(x=prior.sa.sample, y=prior.sa.spline(prior.sa.sample)), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
                  size = dotsize, color = "grey") + ## indicate location of medians
-      ggplot2::geom_point(ggplot2::aes(x, y), data = data.frame(x = prior.sa.sample[median.i], y = prior.sa.spline(prior.sa.sample[median.i])), 
+      ggplot2::geom_point(ggplot2::aes(x=prior.sa.sample[median.i], y=prior.sa.spline(prior.sa.sample[median.i])), data = data.frame(x = prior.sa.sample[median.i], y = prior.sa.spline(prior.sa.sample[median.i])),
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:145]:
```suggestion
      ggplot2::geom_point(ggplot2::aes(x = .data$x, y = .data$y, data = data.frame(x = prior.sa.sample[median.i], y = prior.sa.spline(prior.sa.sample[median.i])), 
```

**Comment by moki1202** [comment]
@infotroph comments resolved.

File: modules/uncertainty/R/plots.R [hunk 11]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
   
   
   trait.plot <- base.plot + ggplot2::ggtitle("Parameter") + 
-    ggplot2::geom_text(ggplot2::aes(y = 1, x = points, label = trait.labels, hjust = 1), size = fontsize$axis/3) + 
+    ggplot2::geom_text(ggplot2::aes(y = 1, x = .data$points, label = trait.labels, hjust = 1), size = fontsize$axis/3) + 
     ggplot2::scale_y_continuous(breaks = c(0, 0), limits = c(0, 1)) + 
     ggplot2::theme(axis.text.x = ggplot2::element_blank())
   cv.plot <- base.plot + ggplot2::ggtitle("CV (%)") + 
-    ggplot2::geom_pointrange(ggplot2::aes(x = points, y = coef.vars, ymin = 0, ymax = coef.vars), size = 1.25) +
+    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$coef.vars, ymin = 0, ymax = plot.data$coef.vars), size = 1.25) +
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:57]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$coef.vars, ymin = 0, ymax = .data$coef.vars), size = 1.25) +
```

File: modules/uncertainty/R/plots.R [hunk 12]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
...
     ggplot2::theme(plot.title = ggplot2::element_text(size = fontsize$title))
   
   el.plot <- base.plot + ggplot2::ggtitle("Elasticity") + 
     ggplot2::theme(plot.title = ggplot2::element_text(size = fontsize$title)) + 
-    ggplot2::geom_pointrange(ggplot2::aes(x = points, y = elasticities, ymin = 0, ymax = elasticities), size = 1.25)
+    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$elasticities, ymin = 0, ymax = plot.data$elasticities), size = 1.25)
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:62]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$elasticities, ymin = 0, ymax = .data$elasticities), size = 1.25)
```

File: modules/uncertainty/R/plots.R [hunk 13]
```diff
@@ -50,20 +50,20 @@ plot_variance_decomposition <- function(plot.inputs,
...
   
   pv.plot <- base.plot + ggplot2::ggtitle("Variance") + 
     ggplot2::theme(plot.title = ggplot2::element_text(size = fontsize$title)) + 
-    ggplot2::geom_pointrange(ggplot2::aes(x = points, sqrt(variances), ymin = 0, ymax = sqrt(variances)), size = 1.25)
+    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, sqrt(.data$variances), ymin = 0, ymax = sqrt(plot.data$variances)), size = 1.25)
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:66]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = sqrt(.data$variances), ymin = 0, ymax = sqrt(.data$variances)), size = 1.25)
```

File: modules/uncertainty/R/plots.R [hunk 14]
```diff
@@ -138,11 +138,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
   if (!is.null(prior.sa.sample) & !is.null(prior.sa.spline)) {
     prior.x <- seq(from = min(prior.sa.sample), to = max(prior.sa.sample), length.out = LENGTH_OUT)
     saplot <- saplot + ## plot spline
-      ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
+      ggplot2::geom_line(ggplot2::aes(x= .data$x, y= .data$y, data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)),
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:141]:
```suggestion
      ggplot2::geom_line(ggplot2::aes(x= .data$x, y= .data$y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
```

File: modules/uncertainty/R/plots.R [hunk 15]
```diff
@@ -138,11 +138,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
   if (!is.null(prior.sa.sample) & !is.null(prior.sa.spline)) {
     prior.x <- seq(from = min(prior.sa.sample), to = max(prior.sa.sample), length.out = LENGTH_OUT)
     saplot <- saplot + ## plot spline
-      ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
+      ggplot2::geom_line(ggplot2::aes(x= .data$x, y= .data$y, data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
                 size = linesize, color = "grey") + ## plot points used to evaluate spline
-      ggplot2::geom_point(ggplot2::aes(x, y), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
+      ggplot2::geom_point(ggplot2::aes(x= .data$x, y= .data$y, data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)),
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:143]:
```suggestion
      ggplot2::geom_point(ggplot2::aes(x= .data$x, y= .data$y), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
```

File: modules/uncertainty/R/plots.R [hunk 16]
```diff
@@ -138,11 +138,11 @@ plot_sensitivity <- function(sa.sample, sa.spline, trait, y.range = c(0, 50), me
...
     prior.x <- seq(from = min(prior.sa.sample), to = max(prior.sa.sample), length.out = LENGTH_OUT)
     saplot <- saplot + ## plot spline
-      ggplot2::geom_line(ggplot2::aes(x, y), data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
+      ggplot2::geom_line(ggplot2::aes(x= .data$x, y= .data$y, data = data.frame(x = prior.x, y = prior.sa.spline(prior.x)), 
                 size = linesize, color = "grey") + ## plot points used to evaluate spline
-      ggplot2::geom_point(ggplot2::aes(x, y), data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
+      ggplot2::geom_point(ggplot2::aes(x= .data$x, y= .data$y, data = data.frame(x = prior.sa.sample, y = prior.sa.spline(prior.sa.sample)), 
                  size = dotsize, color = "grey") + ## indicate location of medians
-      ggplot2::geom_point(ggplot2::aes(x, y), data = data.frame(x = prior.sa.sample[median.i], y = prior.sa.spline(prior.sa.sample[median.i])), 
+      ggplot2::geom_point(ggplot2::aes(x= .data$x, y= .data$y, data = data.frame(x = prior.sa.sample[median.i], y = prior.sa.spline(prior.sa.sample[median.i])),
```
Comment (infotroph) [inline modules/uncertainty/R/plots.R:145]:
```suggestion
      ggplot2::geom_point(ggplot2::aes(x= .data$x, y= .data$y), data = data.frame(x = prior.sa.sample[median.i], y = prior.sa.spline(prior.sa.sample[median.i])), 
```

**Review by infotroph — COMMENTED** [review]
I think I dropped a bunch of parens in my last batch of review comments :(

Will apply them now so we can see the checks pass.

File: modules/data.remote/tests/Rcheck_reference.log [hunk 17]
```diff
@@ -77,6 +77,10 @@ Extensions’ manual.
 Status: 2 WARNINGs, 1 NOTE
 See
   ‘/tmp/Rtmpr2UgRH/PEcAn.data.remote.Rcheck/00check.log’
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (moki1202) [reply modules/data.remote/tests/Rcheck_reference.log]: resolved.

File: modules/uncertainty/R/plots.R (hunk 11 above)
Comment (infotroph) [inline modules/uncertainty/R/plots.R:57]: One more `plot.data` -> `.data` needed -- not sure if I missed this before or if it somehow came back in the other fixes.

```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$coef.vars, ymin = 0, ymax = .data$coef.vars), size = 1.25) +
```

File: modules/uncertainty/R/plots.R (hunk 12 above)
Comment (infotroph) [inline modules/uncertainty/R/plots.R:62]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, y = .data$elasticities, ymin = 0, ymax = .data$elasticities), size = 1.25)
```

File: modules/uncertainty/R/plots.R (hunk 13 above)
Comment (infotroph) [inline modules/uncertainty/R/plots.R:66]:
```suggestion
    ggplot2::geom_pointrange(ggplot2::aes(x = .data$points, sqrt(.data$variances), ymin = 0, ymax = sqrt(.data$variances)), size = 1.25)
```

**Review by mdietze — APPROVED** [review]

**Comment by mdietze** [comment]
@moki1202 SIPNET test failed with an error that's not on our list of usual suspects. One of your changes may have inadvertently changed a data structure slightly

File: modules/uncertainty/R/get.parameter.samples.R [hunk 18]
```diff
@@ -57,16 +57,20 @@ get.parameter.samples <- function(settings,
     ## Load posteriors
     if (!is.na(posterior.files[i])) {
       # Load specified file
-      load(posterior.files[i])
+      env <- new.env()
+      load(posterior.files[i], envir = env)
+      post.distns <- env$post.distns
+      prior.distns <- env$prior.distns
```
Comment (infotroph) [inline modules/uncertainty/R/get.parameter.samples.R:63]: @moki1202 I think this changes some assumptions in the code below: The calls to `exists` below (line 64, 113, 118) are checking whether variables named `post.distns` and `prior.distns` are defined at all, but after this assignment they'll always exist and will be NULL if they weren't present in the loaded `posterior.file`.

The obvious fixes I see are either:
* define the variables conditionally (`if (hasName(env, "post.distns")) post.distns <- env$post.distns `)
* Keep the assignments as they are and replace all `exists(x)` with `!is.null(x)`.
I think I prefer the second option -- it makes it obvious what objects we're expecting the file to contain, while `exists` is hard to reason about and has bitten us in this same way before.

**Pull request merged**